pkg/auth/authz_handler.go

```diff
@@ -51,34 +53,28 @@ func (a *AuthzHandler) CreateAuthorizer() func(*http.Request) error {
 // and the principal was stored in the context in the "AuthKey" context value.
 func (a *AuthzHandler) Authorizer(request *http.Request) error {
 	payload := PayloadFromContext(request.Context())
-	username := payload.Username
-	payloadFromCache, found := a.client.Cache.Get(username)
-	if found {
-		// Update payload with values from cache
-		payloadFromCache := payloadFromCache.(*ocm.AuthPayload)
-		payload.IsUser = payloadFromCache.IsUser
-	} else {
-		// Inquire AMS for user's role
-		allowed, err := a.allowedToUseAssistedInstaller(username)
-		if err != nil {
-			a.log.Errorf("Failed to authorize user: %v", err)
-		} else {
-			payload.IsUser = allowed
-			a.client.Cache.Set(username, payload, cache.DefaultExpiration)
-		}
+	var isAuthorized bool
+	isAuthorized, err := a.allowedToUseAssistedInstaller(payload.Username)
+	if err != nil {
+		return common.NewInfraError(http.StatusInternalServerError, err)
+	} else if !isAuthorized {
+		return common.NewInfraError(
+			http.StatusUnauthorized,
+			fmt.Errorf(
+				"%s: Unauthorized to access route (access review failed)",
+				payload.Username))
 	}
-
-	if payload.IsUser {
-		// authorized user
-		return nil
+	isAuthorized = a.isAuthorizedToAccessRoute(request, payload)
```

Comment:
This should probably be done before calling allowedToUseAssistedInstaller()
since allowedToUseAssistedInstaller involves calling a different service

Reply:
good point I should have noticed that